Documentation/Browser/BrowsingContextsAndNavigables.md

```diff
@@ -0,0 +1,128 @@
+Note that these Environment Records are not actually visible to executing code, and are simply a mechanism
+used by the specification authors to model the language.  Every function and module has a type
+of Declarative Environment Record that contains the variables, functions, catch clauses, and other
+language objects that affect which names are visible at any location in the code. These Environment Records
+are nested, in a tree-like structure that somewhat matches the Abstract Syntax Tree (AST).
+
+The root of the tree of Environment Records is the Global Environment Record, which corresponds to the
+Global Object and its properties. In JavaScript, there is always a ``this`` value representing the current
+object context. At global scope, the Global Object takes that responsibility. In a REPL, that might
+be some REPL specific global object that has global functions to call for doing things like loading
+from the filesystem, or even be as complex as Node or Bun. In a Browser context, the Global object is
+normally the Window, unless there's a Worker involved. This concept is quite different from a native
+executable, where there's no actual object representing the global scope, simply symbols that the
+linker and loader make available to each module.
+
+While the Global Object and its Global Environment represent the root of the tree of identifiers visible
+to the executing JavaScript code, the Global Object isn't sufficient to model all the state around
+a conceptual thread of execution in ECMAScript. This is where the two concepts of [*Realms*](https://tc39.es/ecma262/#sec-code-realms)
+and [*Execution Contexts*](https://tc39.es/ecma262/#sec-execution-contexts) come into play.
+A [*Realm Record*](https://tc39.es/ecma262/#realm-record) is a container that holds a global object,
+its associated Global Environment, a set of intrinsic objects, and any *host* (also called an *embedder*
+in some specification documents) defined extra state that needs to be associated with the realm.
+In LibWeb, the Host Defined slot holds an object that has the HTML Environment Settings Object for each realm,
+as well as all the prototypes, constructors, and namespaces that need to be exposed on the Global Object
+for Web APIs. On top of the Realm abstraction, ECMAScript uses the Execution Context to model the state
+of execution of one particular script or module. On top of the Realm it refers to, the Execution Context
+has information about the current function, the script or module that the current code block belongs to,
+additional Environment Records required to access names in the current scope, any running generator state,
+and most importantly to the thread analogy, the state needed to suspend and resume execution of that script.
+Throughout the ECMAScript specification, the text refers to a stack of execution contexts, with the uppermost
+one always being the [*running execution context*](https://tc39.es/ecma262/#running-execution-context).
```

Comment:
This is relevant because code can only execute if there's at least one context on the stack, otherwise things like "current realm" become ill-defined

Reply:
ignoring the execution context stack for scripts, modules and callbacks seems to be the HTML spec rule rather than the exception, but that's fair, I'll rework this section :)